Spigot-Server-Patches/0230-Max-exp-value-when-merging-orbs.patch

```diff
@@ -0,0 +1,60 @@
+From 14f0b9722752f75b6bd7e9443b5ac1f4bcaa4d68 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 31 Jul 2017 01:35:54 -0500
+Subject: [PATCH] Max exp value when merging orbs
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 54d081fd..18e4581b 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -428,4 +428,10 @@ public class PaperWorldConfig {
+         disableCreeperLingeringEffect = getBoolean("disable-creeper-lingering-effect", false);
+         log("Creeper lingering effect: " + disableCreeperLingeringEffect);
+     }
++
++    public int expMergeMaxValue;
++    private void expMergeMaxValue() {
++        expMergeMaxValue = getInt("exp-merge-max-value", 10);
++        log("Experience Merge Max Value: " + expMergeMaxValue);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index 9dfc2b4d..384592f8 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -1146,16 +1146,25 @@ public abstract class World implements IBlockAccess {
+             EntityExperienceOrb xp = (EntityExperienceOrb) entity;
+             double radius = spigotConfig.expMerge;
+             if (radius > 0) {
++                // Paper start
++                int maxValue = paperConfig.expMergeMaxValue;
+                 List<Entity> entities = this.getEntities(entity, entity.getBoundingBox().grow(radius, radius, radius));
+                 for (Entity e : entities) {
+-                    if (e instanceof EntityExperienceOrb) {
+-                        EntityExperienceOrb loopItem = (EntityExperienceOrb) e;
+-                        if (!loopItem.dead) {
+-                            xp.value += loopItem.value;
+-                            loopItem.die();
+-                        }
++                    if (!(e instanceof EntityExperienceOrb)) {
++                        continue;
++                    }
++                    EntityExperienceOrb loopItem = (EntityExperienceOrb) e;
++                    if (loopItem.dead || loopItem.value >= maxValue) {
++                        continue;
++                    }
++                    xp.value += loopItem.value;
++                    if (xp.value > maxValue) {
++                        loopItem.value = xp.value - maxValue;
++                        continue;
```

Comment:
It would be an exp gain. 

For simplicity I will consider a simplistic case with a maxValue of 100 and a single loopItem.
Initial State: `loopItem.value == 20` and `xp.value == 90`. Total: 110
The loopItem is neither dead nor above the maxValue so it does Not continue.
The loopItem is added to the xp.value.
Current State: `loopItem.value == 20` and `xp.value == 110`. Total: 130.
The xp.value is above the maxValue, so the if statement is followed and sets loopItem.value to (xp.value - maxValue);
Current State: `loopItem.value == 10` and `xp.value == 110`. Total: 120.
Then the loop is continued. With an extra 10 exp.